Source/WebGPU/WebGPU/PresentationContextIOSurface.mm

```diff
@@ -93,12 +93,16 @@
 
 PresentationContextIOSurface::~PresentationContextIOSurface() = default;
 
-void PresentationContextIOSurface::configure(Device&, const WGPUSwapChainDescriptor&)
+void PresentationContextIOSurface::configure(Device& device, const WGPUSwapChainDescriptor&)
 {
+    m_device = &device;
 }
 
 void PresentationContextIOSurface::present()
 {
+    if (m_device)
```

Comment:
Couldn’t we do this without blocking the GPU process main thread?

The reply is async to the web process, is it insufficient to add a completion handler via queue.onSubmittedWorkDone and respond back to the web process at that point?

That would avoid the locks, counters, and blocking the thread